checkbox-provider-ce-oem/bin/multiple-otg.sh

```diff
@@ -0,0 +1,190 @@
+#!/bin/bash
+
+usb_dr_modes=()
+udc_list=()
+mapfile -t usb_dr_modes < <(find "/sys/firmware/devicetree/base/" -name "dr_mode")
+mapfile -t udc_list < <(ls "/sys/class/udc/")
+device_path="/sys/devices/platform/"
+gadget_path="/sys/kernel/config/usb_gadget"
+vid="0xabcd"
+pid="0x1234"
+language="0x409"  # specifically refers to the English language (English-US).
+
+otg_info() {
+    # Mapping following information: 
+    # USB port, USB node, working mode(host/device/otg), UDC
+    IFS=' ' read -ra usb_list <<< "$1"
+    for usb in "${usb_list[@]}"; do
+        IFS=':' read -r port node <<< "$usb"
+        for dr_mode_file in "${usb_dr_modes[@]}"; do
+            # A list of dr_mode in system.
+            usb_node=$(awk -F'[/@]' '{print $(NF-1)}' <<< "$dr_mode_file")
+            otg_mode=$(tr -d '\0' < "$dr_mode_file")
+            # The USB ports and nodes input through the checkbox configuration 
+            # will be mapped to the UDC (USB Device Controller) in the system. 
+            if [ "$node" == "$usb_node" ]; then
+                echo -e "USB_port: $port"
+                echo -e "USB_Node: $usb_node"
+                echo -e "Mode: $otg_mode"
+                found_udc=false
+                for udc in "${udc_list[@]}"; do
+                    if [[ "$udc" == *"$usb_node"* ]] || [[ $(find "$device_path" -wholename "*/$udc/$usb_node.xhci") ]]; then
+                        echo -e "UDC: $udc\n"
+                        found_udc=true
+                        break
+                    fi
+                done
+                if ! "$found_udc"; then
+                    echo -e "UDC: None\n"
+                fi
```

Comment:
it would be more straightforward that we get the udc device like this...
```
udc_dev="None"
for udc in in "${udc_list[@]}"; do
    if [[ "$udc" == *"$usb_node"* ]] || [[ $(find "$device_path" -wholename "*/$udc/$usb_node.xhci") ]]; then
        udc_dev="$udc"
        break
    fi
done
echo -e "UDC: $udc_dev\n"
```